File: ios/Tokopedia/Classses/Object/Shops/ShopPage/ShopProductPageResult.m
//
//  ShopProductPageResult.m
//  Tokopedia
//
//  Created by Johanes Effendi on 3/24/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

#import "ShopProductPageResult.h"

@implementation ShopProductPageResult
+(RKObjectMapping *)mapping{
    RKObjectMapping *shopProductPageResultMapping = [RKObjectMapping mappingForClass:[ShopProductPageResult class]];
    [shopProductPageResultMapping addPropertyMapping:[RKRelationshipMapping relationshipMappingFromKeyPath:@"paging"
                                                                                                 toKeyPath:@"paging"
                                                                                               withMapping:[Paging mapping]]];
    [shopProductPageResultMapping addPropertyMapping:[RKRelationshipMapping relationshipMappingFromKeyPath:@"list"
                                                                                                 toKeyPath:@"list"
                                                                                               withMapping:[ShopProductPageList mapping]]];
    return shopProductPageResultMapping;
}
@end

File: ios/Tokopedia/Features/Object/Shops/ShopPage/ShopProductPageResult.swift
//
//  ShopProductPageResultSwift.swift
//  Tokopedia
//
//  Created by Ferico Samuel on 5/18/17.
//  Copyright © 2017 TOKOPEDIA. All rights reserved.
//

import Unbox

@objc(ShopProductPageResult)
final class ShopProductPageResult: NSObject, Unboxable {
    let paging: Paging?
    let list: [ShopProductPageList]

    init(paging: Paging?, list: [ShopProductPageList]) {
        self.paging = paging
        self.list = list
    }

    convenience init(unboxer: Unboxer) throws {
        self.init(
            paging: try? unboxer.unbox(keyPath: "data.paging") as Paging,
            list: try unboxer.unbox(keyPath: "data.list")
        )
    }
}
